core/controllers/suggestion.py

```diff
@@ -344,8 +352,27 @@ def get(self, target_type, suggestion_type):
         """
         self._require_valid_suggestion_and_target_types(
             target_type, suggestion_type)
+        topic_name = self.request.get('topic_name', None)
+
+        opportunity_summary_exp_ids_specific_to_topic = None
+        if topic_name is not None:
+            topic = topic_fetchers.get_topic_by_name(topic_name)
+            if topic is not None:
+                exploration_opportunity_summaries = (
+                    opportunity_services.
+                    get_exploration_opportunity_summaries_by_topic_id(
+                        topic.id))
+
+                opportunity_summary_exp_ids_specific_to_topic = [
+                    opportunity.id for opportunity
+                    in exploration_opportunity_summaries]
+            elif topic_name != 'All':
```

Comment:
Changes this from else: to elif: to not raise this error when the topic_name sent back is 'All'